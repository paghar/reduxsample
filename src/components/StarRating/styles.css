.star-rating {
  display: flex;
  flex-wrap: wrap;
}

.star {
  cursor: pointer;
  width: 1.5em;
  height: 1.5em;
  background-color: #ff80b3;
  -webkit-clip-path: polygon(
    50% 0%,
    63% 38%,
    100% 38%,
    69% 59%,
    82% 100%,
    50% 75%,
    18% 100%,
    31% 59%,
    0% 38%,
    37% 38%
  );
  clip-path: polygon(
    50% 0%,
    63% 38%,
    100% 38%,
    69% 59%,
    82% 100%,
    50% 75%,
    18% 100%,
    31% 59%,
    0% 38%,
    37% 38%
  );
}

.star.selected {
  background-color: #e6005c;
}
