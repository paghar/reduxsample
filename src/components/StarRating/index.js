import React from "react"
import "./styles.css"

const Star = ({ selected = false, onClick = f => f }) => (
  <div className={selected ? "star selected" : "star"} onClick={onClick} />
)

const StarRating = ({ totalStars, rateValue, clicked }) => {
  return (
    <div className="star-rating">
      {[...Array(totalStars)].map((n, i) => (
        <Star key={i} selected={i < rateValue} onClick={() => clicked(i + 1)} />
      ))}
    </div>
  )
}

export default StarRating
