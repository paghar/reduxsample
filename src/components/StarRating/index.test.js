import React from "react"
import renderer from "react-test-renderer"
import StarRating from "."
import { shallow, mount } from "enzyme"
describe("<StarRating />", () => {
  it("StarRating", () => {
    const tree = renderer
      .create(<StarRating totalStars="5" rateValue="3" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("StarRating Click Event", () => {
    const mockCallBack = jest.fn()
    const wrapper = mount(
      <StarRating totalStars="5" rateValue="3" clicked={mockCallBack} />
    )

    wrapper
      .find("div")
      .at(1)
      .simulate("click")
    expect(mockCallBack).toHaveBeenCalled()
  })
})
