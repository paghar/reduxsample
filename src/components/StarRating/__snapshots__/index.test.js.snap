// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> StarRating 1`] = `
<div
  className="star-rating"
>
  <div
    className="star selected"
    onClick={[Function]}
  />
</div>
`;
